fe/fe-core/src/main/java/com/starrocks/sql/ast/TableRelation.java

```diff
@@ -53,6 +54,8 @@ public enum TableHint {
 
     private Map<Expr, SlotRef> generatedExprToColumnRef = new HashMap<>();
 
+    private List<String> guessScanColumns = Collections.emptyList();
```

Comment:
I think scan columns  should be determinate,  not guessed.  could we rename it to prunedScanColumns?  which is consistent with your session variable name:  enable_analyze_phase_prune_columns

Reply:
not all case can be pruned, we can't compute the complate relation now, just can be guess....